### DISCUSSION TITLE
Cloud Run support for Serverpod

### DISCUSSION THREAD

**Discussion by ajay-k07** [serverpod/serverpod#508] [Q&A]
Hi all im trying to create a GCP workflow for serverpod as we have for AWS to contribute soon will be create document and video tutorial . i have completed everything and its working file but the problem is im getting database related error. Even if my database is running 

```
[
  {
    "textPayload": "CPU and memory usage metrics are not supported on this platform."
  },
  {
    "textPayload": "2022-12-24 16:54:20.719025Z Internal server error. Failed to connect to database in future call manager."
  },
  {
    "textPayload": "TimeoutException after 0:00:15.000000: Future not completed"
  },
  {
    "textPayload": "===== asynchronous gap ==========================="
  },
  {
    "textPayload": "package:postgres_pool/postgres_pool.dart 331                 PgPool.run.<fn>"
  },
  {
    "textPayload": "package:retry/retry.dart 131                                 RetryOptions.retry"
  },
  {
    "textPayload": "package:postgres_pool/postgres_pool.dart 329                 PgPool.run"
  },
  {
    "textPayload": "package:serverpod/src/database/database_connection.dart 443  DatabaseConnection.deleteAndReturn"
  },
  {
    "textPayload": "package:serverpod/src/database/database.dart 166             Database.deleteAndReturn"
  },
  {
    "textPayload": "package:serverpod/src/server/future_call_manager.dart 100    FutureCallManager._checkQueue"
  },
  {
    "textPayload": "#0      FutureCallManager._checkQueue (package:serverpod/src/server/future_call_manager.dart:136)"
  }
]
```

got the same error in local after leaving the serverpod server ideal 

```

SERVERPOD version: 0.9.21 mode: development time: 2022-12-24 17:54:05.964509Z
Insights listening on port 8081
Server default listening on port 8080
Webserver listening on port 8082
METHOD CALL: example.hello duration: 5ms numQueries: 0 authenticatedUser: null
METHOD CALL: book.save duration: 783ms numQueries: 0 authenticatedUser: null
METHOD CALL: example.hello duration: 0ms numQueries: 0 authenticatedUser: null
METHOD CALL: book.save duration: 115ms numQueries: 0 authenticatedUser: null
2022-12-24 19:02:05.776521Z Internal server error. Failed to connect to database in future call manager.
SocketException: Connection reset by peer (OS Error: Connection reset by peer, errno = 54), address = 144.24.151.239, port = 58955
===== asynchronous gap ===========================
package:postgres_pool/postgres_pool.dart 331:18                 PgPool.run.<fn>
package:retry/retry.dart 131:16                                 RetryOptions.retry
package:postgres_pool/postgres_pool.dart 329:14                 PgPool.run
package:serverpod/src/database/database_connection.dart 443:20  DatabaseConnection.deleteAndReturn
package:serverpod/src/database/database.dart 166:12             Database.deleteAndReturn
package:serverpod/src/server/future_call_manager.dart 100:18    FutureCallManager._checkQueue

Local stacktrace:
#0      FutureCallManager._checkQueue (package:serverpod/src/server/future_call_manager.dart:136:36)
<asynchronous suspension>


```

I know it may feel like some kind of postgress issue but want to know is there any serverpod related which i can work on 

There is something happening with package:serverpod/src/server/future_call_manager.dart 100 FutureCallManager._checkQueue

sorry if its out of topic but i feel like if we are able to support top cloud providers it will be really helpful.I used official dart image to build the docker image  and please take a look at my code repo 

[Repo](https://github.com/ajay-k07/serverpod_cloudrun)

**Comment by yahu1031** [comment]
> "textPayload": "CPU and memory usage metrics are not supported on this platform."

As we don't have enough support for GCP, That is why you seeing these issues. And while going through the docs, you will be seeing a warning statement that we don't have enough support for GCP as we have for AWS.

**Reply by ajay-k07** [reply]
i dont have issues with  CPU and memory usage metrics are not supported on this platform.

**Reply by yahu1031** [reply]
then? Please share the whole stack traces.

**Reply by ajay-k07** [reply]
actually im getting this only others are just timestamp

**Reply by ajay-k07** [reply]
but the server is up and running im able to connect and get response from the hello world end point

**Reply by ajay-k07** [reply]
> then? Please share the whole stack traces.

{
  "textPayload": "Local stacktrace:",
  "insertId": "63a6d90a000231854352be33",
  "resource": {
    "type": "cloud_run_revision",
    "labels": {
      "configuration_name": "serverpod-demo",
      "revision_name": "serverpod-demo-00002-fem",
      "location": "asia-south1",
      "service_name": "serverpod-demo",
      "project_id": "serverpod"
    }
  },
  "timestamp": "2022-12-24T10:48:42.143749Z",
  "logName": "projects/serverpod/logs/run.googleapis.com%2Fstderr",
  "receiveTimestamp": "2022-12-24T10:48:42.147676136Z"
}


nothing is logged

**Reply by yahu1031** [reply]
Not aware of it to date. Maybe @vlidholt can help you.

**Reply by ajay-k07** [reply]
ok thanks no problem

**Comment by ajay-k07** [comment]
@vlidholt Please let me know if we can merge this workflow or do you have any suggestion.Suppose if It can be merged i can work on the documentation part and tutorial part  😄

**Comment by yahu1031** [comment]
@ajay-k07 Just create a PR and wait for the approval.

**Reply by ajay-k07** [reply]
do have to work on the generator part so that we can add workflow support out of the box ?.Please guide me on this

**Reply by yahu1031** [reply]
Please let us know at what point you would need help!

**Reply by ajay-k07** [reply]
if i just add this files in inside the [serverpod_templates](https://github.com/serverpod/serverpod/tree/main/templates/serverpod_templates) will it get added to the projects when we create a new projects

**Comment by yahu1031** [comment]
Yes it does

**Reply by ajay-k07** [reply]
ok thanks will do the same
